src/borg/testsuite/archive_test.py

```diff
@@ -402,3 +407,134 @@ def test_reject_non_sanitized_item():
     for path in rejected_dotdot_paths:
         with pytest.raises(ValueError, match="unexpected '..' element in path"):
             Item(path=path, user="root", group="root")
+
+
+def test_compare_and_extract_chunks(tmpdir, monkeypatch):
```

Comment:
maybe use less c&p for the tests. the code in these is very similar, so you could replace it by generic code and parametrize that.